components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
> But imagine the case where a user is typing just slow enough to trigger multiple searches- they'll have content appearing and disappearing seemingly at random.

Um I guess that's the nature of search-as-you-type. We don't start new search on every keystroke so to me that's fair enough. For non slower typers having previous search result stayed can make it look like the system is having problem detecting the new keyword user just typed, especially when network connection is bad.

Anyways, since you two both agree on the same approach I'll update my PR 😄